cmd/maya-exporter/app/collector/zvol/pool_synctime.go

```diff
@@ -110,6 +115,12 @@ func (p *poolMetrics) get() *poolfields {
 // Collect is implementation of prometheus's prometheus.Collector interface
 func (p *poolMetrics) Collect(ch chan<- prometheus.Metric) {
 	p.Lock()
+	if p.isRequestInProgress() {
+		p.zpoolListRequestRejectCounter.Inc()
+		p.Unlock()
+		p.zpoolListRequestRejectCounter.Collect(ch)
```

Comment:
What will the above line (line no. 121) do exactly?

Reply:
Collect is called by the Prometheus registry when collecting
metrics. The implementation sends each collected metric via the
provided channel and returns once the last metric has been sent. The
descriptor of each sent metric is one of those returned by Describe
(unless the Collector is unchecked, see above). Returned metrics that
share the same descriptor must differ in their variable label values.

The above statement is taken Prometheus docs/comments.

Reply:
https://github.com/prometheus/client_golang/blob/110f6d46931dec63add7ad91e503224d764f1e9a/prometheus/collector.go#L50